Api/WebhookController.cs

```diff
@@ -0,0 +1,146 @@
+
+[ApiController]
+[Route("api/[controller]")]
+public class WebhookController : ControllerBase
+{
+    private readonly ILogger<WebhookController> logger;
+    private readonly HttpClient httpClient;
+    private readonly string clientId, clientSecret, requestTokenBaseUrl, calendarUrl;
+
+    public WebhookController(IConfiguration configuration, ILogger<WebhookController> logger, HttpClient httpClient)
+    {
+        this.logger = logger;
+        this.httpClient = httpClient;
+        clientId = configuration["GoogleApi:ClientID"];
+        clientSecret = configuration["GoogleApi:Secret"];
+        requestTokenBaseUrl = configuration["GoogleApi:RequestTokenUrl"];
+        calendarUrl = configuration["GoogleApi:CalendarUrl"];
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> IncomingMessage()
+    {
+        string messageBody = Request.Form["Body"];
+        string messageFrom = Request.Form["From"];
+
+        var userCalendarConfig = UserCalendarConfiguration.Instance;
+
+        // Extract appointment
+        var utcOffset = TimeZoneInfo.Local.BaseUtcOffset;
+        var timeZone = "UTC" + (utcOffset > TimeSpan.Zero ? "+" : "-") + utcOffset.ToString(@"h\:mm");
+        var appointmentDetails = AppointmentDetailsExtractor.Extract(messageBody, timeZone, messageFrom);
+
+        // Check days and time 
+        var dateTimeOpen = DateTime.ParseExact(userCalendarConfig.OpeningTime, "HH:mm", CultureInfo.InvariantCulture);
+        var dateTimeClose = DateTime.ParseExact(userCalendarConfig.ClosingTime, "HH:mm", CultureInfo.InvariantCulture);
+
+        var dayOfWeek = appointmentDetails.Start.DateTime.DayOfWeek.ToString();
+
+        if (!userCalendarConfig.CheckedDays.Split(",").Contains(dayOfWeek))
+        {
+            var response = new MessagingResponse();
+            response.Message($"Business does not take appointment on {dayOfWeek}");
+            return new TwiMLResult(response);
+        }
+
+        if (!(appointmentDetails.Start.DateTime.TimeOfDay >= dateTimeOpen.TimeOfDay &&
+              appointmentDetails.End.DateTime.TimeOfDay <= dateTimeClose.TimeOfDay))
+        {
+            return new MessagingResponse()
+                .Message($"Business only takes appointment between {dateTimeOpen.TimeOfDay} and {dateTimeClose.TimeOfDay}")
+                .ToTwiMLResult();
+        }
+
+        var accessToken = await GetAccessToken(userCalendarConfig);
+
+        // Insert event into the Google Calendar
+        var requestBody = new StringContent(JsonConvert.SerializeObject(appointmentDetails));
+        httpClient.DefaultRequestHeaders.Authorization =
+            new AuthenticationHeaderValue("Bearer", accessToken);
+
+        // Check if no event exist in that date and time
+        var appointmentDate = appointmentDetails.Start.DateTime.Date;
+        var eventsForDayUrl = calendarUrl +
+                              $"?timeMin={XmlConvert.ToString(appointmentDate, XmlDateTimeSerializationMode.Utc)}" +
```

Comment:
XML convert converts the datetime into the right format as Google Calendar API expects it.